tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/worker/WaitForAnyActionsTest.java

```diff
@@ -172,44 +173,38 @@ public void waitTest17() {
     public void waitTest18() {
         Object vals = BRunUtil.invoke(result, "waitTest18", new Object[0]);
 
-        Assert.assertEquals("65", vals.toString());
-    }
-
-    @Test
-    public void waitTest19() {
-        BRunUtil.invoke(result, "waitTest19", new Object[0]);
+        Assert.assertTrue(Arrays.asList("60", "65").contains(vals.toString()));
     }
 
     @Test
     public void waitTest20() {
-        Object vals = BRunUtil.invoke(result, "waitTest20", new Object[0]);
-
-        Assert.assertEquals("hello moo", vals.toString());
+        try {
+            Object vals = BRunUtil.invoke(result, "waitTest20", new Object[0]);
+            Assert.assertEquals("hello moo", vals.toString());
+        } catch (RuntimeException e) {
+            Assert.assertTrue(e.getMessage().contains("err returned"));
+        }
     }
 
     @Test(expectedExceptions = {BLangRuntimeException.class},
-            expectedExceptionsMessageRegExp = "error: A hazardous error occurred!!! Panic!!.*")
+            expectedExceptionsMessageRegExp = "error: A hazardous error occurred!!!.*")
     public void waitTest21() {
         BRunUtil.invoke(result, "waitTest21", new Object[0]);
     }
 
-    @Test
-    public void waitTest22() {
-        Object vals = BRunUtil.invoke(result, "waitTest22", new Object[0]);
-
-        Assert.assertEquals("hello foo", vals.toString());
+    @Test(dataProvider = "waitForAnyActionsTestFunctions")
+    public void testWaitForAnyActionsTest(String functionName) {
+        BRunUtil.invoke(result, functionName);
     }
 
-    @Test
-    public void waitTest23() {
-        Object vals = BRunUtil.invoke(result, "waitTest23", new Object[0]);
-
-        Assert.assertNull(vals);
-    }
-
-    @Test
-    public void waitTest24() {
-        BRunUtil.invoke(result, "waitTest24", new Object[0]);
+    @DataProvider
```

Comment:
Shall we move the validations of other tests to the ballerina side?

Reply:
Also removed 3 duplicate tests.